sale_product_set/models/res_config.py

```diff
@@ -0,0 +1,11 @@
+from odoo import fields, models
+
+
+class ResConfigSettings(models.TransientModel):
+    _inherit = "res.config.settings"
+
+    archive_partner_product_sets = fields.Boolean(
+        string="Archive product sets together with partner",
+        config_parameter="sale.archive_partner_product_sets",
```

Comment:
```suggestion
        config_parameter="sale_product_set.archive_partner_product_sets",
```
Not that is strictly required, but it avoids confusion with potential keys from sale module